Many great things!

OliverEvans96 opened the pull request:
Hi friend,

I made a lot of changes, including:
- Describe the conda environment with `environment.yml`, which greatly simplifies package management and enables the use of Binder (see below)
- Write (okay, copy from StackOverflow and modify) some code (download_progress.py) to download the data files from within the notebook (feat. progress bars!)
- Clean up directory handling. Choose one data folder, and put all the inputs and outputs there. I was trying to have a separate output folder, but `find_event_pipeline` seems to require the `.dat` and `.h5` files to be in the same directory, and `plot_event_pipeline` seems to create the output images in the same directory as the `.dat` files. So having a single directory is basically the only option.
- Add a link to the README to Binder, which creates a private Jupyter server in a google datacenter somewhere for anyone who wants to run the notebook from their web browser. The download speeds are super fast (Google's network is pretty great), but they only allocate a very small amount of processing power to the each person's private server, so the turboSETI calculations run pretty slowly (~5min). **NOTE**: you have to change the link in the README to point to your repo instead of mine once you merge the changes.
- Miscellaneous code cleanup

Hopefully I haven't messed anything up while making all these changes. Let me know when you want to talk more.

Good luck!
Oliver

OliverEvans96 commented:
Oh, one thing that needs to be changed: [I'm currently using my fork of turboSETI](https://github.com/OliverEvans96/VoyagerTutorial2/blob/master/environment.yml#L17) due to a bug in the official installation script. I submitted a patch (https://github.com/UCBerkeleySETI/turbo_seti/pull/183), but until it's accepted, the `environment.yml` will only work with my fork. It's a pretty minor change, so hopefully it won't take long. At that point, we should switch back to using the official version of the code.

texadactyl commented:
@OliverEvans96
Have you tested on MacOS?  Make sure that it works there as @elanlavie is wworking in that environment.

OliverEvans96 commented:
Thanks for merging the change @texadactyl, and good idea to check on mac. It is working on mac, but I discovered and fixed a bug I introduced that pops up when jupyter widgets aren't available. Should be all good now!

texadactyl commented:
I never ran into a jupyter issue on MacOS.  Then again, I discarded conda and just used straight ```pip``` for installing packages.  Everything worked immediately wwithout issues.

Here is precisely what I did:
* I installed Python3 for MacOS.
* In the home directory, I created a directory "seti" and under that a subdirectory called "voyager_2020". 
* I put all of the 6 Voyager .h5 files in voyager_2020. 
* Directly under "seti", I placed 3 files:  VoyagerTutorial.ipynb (Elan's notebook file), install.sh (a bash script for installing necessary packages), and test1.py (a Python test program).  See ```zippy.zip``` attachment.
* If you copy what I did, make sure to verify the directory name ("DIR") in test1.py.  It should be okay.  Essentially, test1 runs turbo_seti on the first .h5 in the voyager_2020 subdirectory. Confidence builder.
* I opened a terminal window and executed cd seti. 
* Then, I ran ```python3 test1.py``` .  That worked as expected.
* Then, I ran ```jupyter notebook```.
* In the notebook, I browsed to the seti directory and double-clicked on the notebook file.
I had to edit the value of "directory_constant" (In [2]) to point to the right place on the Mac.
* Then, I clicked on Run all the way to the end of the notebook file.  Success!


[zippy.zip](https://github.com/elanlavie/VoyagerTutorialRepository/files/5984325/zippy.zip)

texadactyl commented:
@elanlavie The previous message should be a repeat of what I stated before on slack.
